### ISSUE TITLE
Empty require in ns doesn't throw an error

### ISSUE THREAD

**Issue by NoahTheDuke** [clj-kondo/clj-kondo#2416]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2024.08.29

**macro usage**

N/A

**platform**

All

**editor**

N/A

**problem**

```clojure
(ns foo (:require))
```

isn't an error.

**repro**

```
$ echo "(ns foo (:require))" | clj-kondo --lint -
linting took 6ms, errors: 0, warnings: 0
```

**config**

N/A

**expected behavior**

```
$ echo "(ns foo (:require))" | clj-kondo --lint -
<stdin>:1:19: error: require form is invalid: form must not be empty
linting took 5ms, errors: 1, warnings: 0
```

If this is accepted, I can write the PR to add it.

**Comment by borkdude** [comment]
Yeah, maybe `Nothing specified to load` should be the message, which Clojure JVM shows?

**Comment by borkdude** [comment]
ClojureScript does support an empty `:require`

**Comment by NoahTheDuke** [comment]
I get a spec error when it's in an `ns` form:

```
$ cat test/clojure/core_test/some_q.cljc
(ns clojure.core-test.some-q
  (:require))

$ lein test
Syntax error macroexpanding clojure.core/ns at (clojure/core_test/some_q.cljc:1:1).
((:require)) - failed: Extra input spec: :clojure.core.specs.alpha/ns-form
```

I can add a check for the `require` function too, if you'd like.

**Comment by borkdude** [comment]
sure! I guess the exact warning doesn't matter so much as long as it's clear.

**Comment by scottbale** [comment]
This form now yields a lint error, but should it?

```clj
(ns tmp
  (:require #?(:clj [schema.core :as s])
            #?(:clj [schema.utils :as sutils])))
```

```
% clj-kondo --lint tmp.cljc
tmp.cljc:2:4: error: Invalid require: no libs specified to load
tmp.cljc:2:22: warning: namespace schema.core is required but never used
tmp.cljc:3:22: warning: namespace schema.utils is required but never used
linting took 13ms, errors: 1, warnings: 2
% clj-kondo --version
clj-kondo v2025.01.16
```

I guess the error is because the `:require` would be empty in the ClojureScript case, but as @borkdude said above:
> ClojureScript does support an empty `:require`

I can work around by doing this:
```clj
(ns tmp
  #?(:clj (:require #?(:clj [schema.core :as s])
                    #?(:clj [schema.utils :as sutils]))))
```
